test/algorithms/distribution_learners/qgan/test_qgan.py

```diff
@@ -377,6 +377,42 @@ def test_qgan_gradient_penalty_pytorch(self):
         )
         self.assertAlmostEqual(trained_qasm["rel_entr"], trained_statevector["rel_entr"], delta=0.1)
 
+    @data("param_shift", "fin_diff", "lin_comb")
+    def test_qgan_training_custom_gradients_statevector(self, gradient_method: str):
+        """
+        Test qGAN different gradient methods for the statevector simulator
+
+        Args:
+            gradient_method (str): custom gradient method
+        """
+        if gradient_method is None:
+            gradient_method = "param_shift"
+
```

Comment:
This code seems unnecessary. The typehint in the method says just 'str' and does not include Optional. All values are passed to the test by ddt anyway so it should never be done unless someone changes something. I think if you wanted to guard against someone altering the ddt params or whatever then instead do a self.skipTest("gradient method must be provided") or something like this. Beyond if you want to default gradient_method to "param_shift" I think it would be better just to put it in the parameter as a default value.